src/tmx/Messages/include/SensorDetectedObject.h

```diff
@@ -28,23 +28,115 @@ namespace tmx
             static constexpr const char *MessageSubType = MSGSUBTYPE_SENSOR_DETECTED_OBJECT_STRING;
 
             // TODO: Convert this member variable to std::attributes and handle nested object and arrays. (see [CloudHeartbeatMessage.h](./CloudHearbeatMessage.h) array_attribute )
-            
-            // Classification of detected object
-            std::string type = "";
+            //Flag to indicate whether sensor detected object is simulated.
+            std_attribute(this->msg, bool, ISSimulated, false,);
+            // Classification of detected object.
+            std_attribute(this->msg, std::string, Type, "",);
             // Confidence of type classification
-            double confidence = 0.0;
-            // Unique indentifier of sensor reporting detection
-            std::string sensorId = "";
-            // String describing projection used to convert cartesian data to WGS84 data 
-            std::string projString = "";
-            // Unique identifier of detected object
-            int objectId = 0;
+            std_attribute(this->msg, double, Confidence, 0.0,);
+            // Unique indentifier of sensor reporting detection.
+            std_attribute(this->msg, std::string, SensorId, "", );
+            // String describing projection used to convert cartesian data to WGS84 data.
+            std_attribute(this->msg, std::string, ProjString, "", );
+            // Unique identifier of detected object.
+            std_attribute(this->msg, int, ObjectId, 0, );
+
             // Cartesian positiion of object. Assumed to be ENU coordinate frame.
-            tmx::utils::Point position = tmx::utils::Point();
+            typedef struct Position{
```

Comment:
It may be worth while creating separate .h files for some of these classes in the messages package. This will allow reuse of these types in future messages which I think it pretty likely. It also makes  this class more readable since it will not include object definitions of nest objects amongst its own member declaration